QuestionBot/QuestionBot/Model/InMemoryStore.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+
+    public class InMemoryStore : IStore {
+        private IList<IRecord> allRecords = new List<IRecord>();
+        private static int _questionId = 0;
+
+        public IRecord CreateRecord(string question){
+            _questionId++;
+
+            Record newQuestion = new Record(_questionId, question, DateTime.Now);
+            allRecords.Add(newQuestion);
+
+            return newQuestion;
+        }
+
+        public bool TryUpdateRecord(int id, string answer, out IRecord recordToUpdate){
```

Comment:
By convention, `Try` methods should not throw. Can you spot the line in this file that *could* potentially throw?